### DISCUSSION TITLE
Is there a way to disable Dropdown

### DISCUSSION THREAD

**Discussion by powertec-dan** [vikramlearning/blazorbootstrap#371] [Q&A] [answered]
There appears to be a way to disable all the DropDownItems, but I'd like to show the dropdown itself as unavailable.

I tried to add Disabled="@value" to the DropdownToggleButton but the dropdown stops functioning after any other event is thrown on the page.

Any help or advice would be appreciated.

Cheers,
Dan

**Comment by gvreddy04** [comment, marked as answer]
@powertec-dan At this moment Dropdown disable is not supported.

**Reply by powertec-dan** [reply]
Any chance of requesting it as a feature? :)

**Reply by gvreddy04** [reply]
@powertec-dan Please create an issue, and I will add the feature. For now, I can share a simple workaround here.

## Example:

```cshtml
<Dropdown>
    <DropdownToggleButton Color="ButtonColor.Secondary" disabled="@disabled">Dropdown button</DropdownToggleButton>
    <DropdownMenu>
        <DropdownItem To="#" Type="ButtonType.Link">Action</DropdownItem>
        <DropdownItem To="#" Type="ButtonType.Link">Another action</DropdownItem>
        <DropdownItem To="#" Type="ButtonType.Link">Something else here</DropdownItem>
    </DropdownMenu>
</Dropdown>

@code
{
    private bool disabled = true;
}
```

## Screenshot:

![image](https://github.com/vikramlearning/blazorbootstrap/assets/2337067/d37241d4-e753-45c5-9221-12fbe96ee35c)

**Reply by powertec-dan** [reply]
I've raised an issue.

Thanks for posting the workaround. 

However, the problem with this workaround is that when the Dropdown becomes enabled again, the drop list no longer appears when you click on the Dropdown. So, while the control looks like it is enabled, it doesn't function properly.

This is what I experienced when I tried something very similar earlier in the day.

Cheers,
Dan

**Reply by gvreddy04** [reply]
@powertec-dan This information is helpful, I'll consider this scenario.

**Comment by gvreddy04** [comment]
@powertec-dan I do not recommend this approach, but if you have a mandatory scenario for disabling a dropdown, you can use two dropdowns with an if-else condition and the html disabled attribute as an interim solution.

**Reply by powertec-dan** [reply]
I will consider that. For the moment I can live with the dropdowns being enabled. I have a few months before these user interfaces need to be customer facing. So I would prefer to wait for a less temporary solution.
